### PR TITLE
Spring_Rest_Practise_HW

### PR DESCRIPTION
**Pull request by AlexPravda1**
- OrderController 
-- Complete order
-- Get orders history for user 

- AuthenticationController
-- Register 

- UserController
-- Get user by email 

- ShoppingCartController
-- Add movie session
-- Get by user

### CODE REVIEW

**Review by Nick97-git — CHANGES_REQUESTED** [review]
Let's fix this first:
Screenshot 2022-04-14 at 08 19 01

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeOrder(
+            @RequestParam(value = "userId") Long userId) {
+        ShoppingCart shoppingCartByUser = shoppingCartService.getByUser(userService.get(userId));
+        return orderDtoResponseMapper.toDto(orderService.completeOrder(shoppingCartByUser));
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getOrdersDtoHistory(
```
Comment (Nick97-git) [inline src/main/java/mate/academy/spring/controller/OrderController.java:44]:
```suggestion
    public List<OrderResponseDto> getOrdersHistory(
```

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.orderDtoResponseMapper = orderDtoResponseMapper;
+        this.userService = userService;
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeOrder(
+            @RequestParam(value = "userId") Long userId) {
+        ShoppingCart shoppingCartByUser = shoppingCartService.getByUser(userService.get(userId));
```
Comment (Nick97-git) [inline src/main/java/mate/academy/spring/controller/OrderController.java:39]:
```suggestion
        ShoppingCart shoppingCart = shoppingCartService.getByUser(userService.get(userId));
```

File: src/main/java/mate/academy/spring/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                          DtoResponseMapper<UserResponseDto, User> userDtoResponseMapper) {
+        this.userService = userService;
+        this.userDtoResponseMapper = userDtoResponseMapper;
+    }
+
+    @GetMapping("/email")
+    public UserResponseDto getUserByEmail(
+            @RequestParam(value = "email") String email)
+            throws AuthenticationException {
+        Optional<User> userByEmail = userService.findByEmail(email);
```
Comment (Nick97-git) [inline src/main/java/mate/academy/spring/controller/UserController.java:30]:
```suggestion
        Optional<User> user = userService.findByEmail(email);
```

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @PutMapping("/movie-sessions")
+    public ShoppingCartReponseDto addMovieSessionToShoppingCart(
+            @RequestParam(value = "userId") Long userId,
+            @RequestParam(value = "movieSessionId") Long movieSessionId) {
+        User user = userService.get(userId);
+        MovieSession movieSession = movieSessionService.get(movieSessionId);
+        shoppingCartService.addSession(movieSession, user);
+
```
Comment (Nick97-git) [inline src/main/java/mate/academy/spring/controller/ShoppingCartController.java:50]: remove empty line

File: src/main/java/mate/academy/spring/model/dto/response/OrderResponseDto.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
+package mate.academy.spring.model.dto.response;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long orderId;
+    private List<Long> ticketsId;
```
Comment (Nick97-git) [inline src/main/java/mate/academy/spring/model/dto/response/OrderResponseDto.java:8]:
```suggestion
    private List<Long> ticketIds;
```

File: src/main/java/mate/academy/spring/model/dto/response/ShoppingCartReponseDto.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package mate.academy.spring.model.dto.response;
+
+import java.util.List;
+
+public class ShoppingCartReponseDto {
+    private Long cartId;
+    private List<Long> ticketsId;
```
Comment (Nick97-git) [inline src/main/java/mate/academy/spring/model/dto/response/ShoppingCartReponseDto.java:7]:
```suggestion
    private List<Long> ticketIds;
```

**Review by Nick97-git — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java (hunk 4 above)
Comment (AlexPravda1) [reply src/main/java/mate/academy/spring/controller/ShoppingCartController.java:50]: The intention was to get visual readability.
Line removed, thank you

Comment (Nick97-git) [reply src/main/java/mate/academy/spring/controller/ShoppingCartController.java]: when we talk about small methods it doesn't make sense, I could understand if you'd have method with 20 lines of code than may be it'd make sense, but even in such case it'd be better to split such into several small ones, so you won't need to add these empty lines

**Review by Nick97-git — APPROVED** [review]
Answered to your comment

**Pull request closed without merging**